src/NuGet.Core/NuGet.DependencyResolver.Core/ResolverUtility.cs

```diff
@@ -215,27 +219,26 @@ public static class ResolverUtility
         /// </summary>
         /// <param name="cache">Cache of requests per library</param>
         /// <param name="libraryRange">The library requested</param>
-        /// <param name="remoteProviders">remote Providers (all sources, including file sources)</param>
-        /// <param name="localProviders">local providers(gpf, fallback folders)</param>
-        /// <param name="cacheContext">source caching context</param>
+        /// <param name="remoteWalkContext">remote Providers (all sources, including file sources)</param>
         /// <param name="logger">logger</param>
         /// <param name="cancellationToken">cancellation token</param>
         /// <returns>The requested range and remote match.</returns>
         public static Task<Tuple<LibraryRange, RemoteMatch>> FindPackageLibraryMatchCachedAsync(
             ConcurrentDictionary<LibraryRange, Task<Tuple<LibraryRange, RemoteMatch>>> cache,
             LibraryRange libraryRange,
-            IEnumerable<IRemoteDependencyProvider> remoteProviders,
-            IEnumerable<IRemoteDependencyProvider> localProviders,
-            SourceCacheContext cacheContext,
+           RemoteWalkContext remoteWalkContext,
             ILogger logger,
             CancellationToken cancellationToken)
         {
-            return cache.GetOrAdd(libraryRange, (cacheKey) => ResolvePackageLibraryMatchAsync(libraryRange, remoteProviders, localProviders, cacheContext, logger, cancellationToken));
+            return cache.GetOrAdd(libraryRange, (cacheKey) => ResolvePackageLibraryMatchAsync(libraryRange, remoteWalkContext, logger, cancellationToken));
         }
 
-        private static async Task<Tuple<LibraryRange, RemoteMatch>> ResolvePackageLibraryMatchAsync(LibraryRange libraryRange, IEnumerable<IRemoteDependencyProvider> remoteProviders, IEnumerable<IRemoteDependencyProvider> localProviders, SourceCacheContext cacheContext, ILogger logger, CancellationToken cancellationToken)
+        private static async Task<Tuple<LibraryRange, RemoteMatch>> ResolvePackageLibraryMatchAsync(LibraryRange libraryRange, RemoteWalkContext remoteWalkContext, ILogger logger, CancellationToken cancellationToken)
         {
-            var match = await FindPackageLibraryMatchAsync(libraryRange, NuGetFramework.AnyFramework, remoteProviders, localProviders, cacheContext, logger, cancellationToken);
+            IList<IRemoteDependencyProvider> remoteDependencyProviders = remoteWalkContext.FilterDependencyProvidersForLibrary(libraryRange);
```

Comment:
This code path is invoked while restoring `PackageDownload` items.

Reply:
Do you mean download package from uri?

Reply:
I am referring to `PackageDownload` items as per https://github.com/NuGet/Home/wiki/%5BSpec%5D-PackageDownload-support spec